python_modules/dagster/dagster/_cli/code_server.py

```diff
@@ -0,0 +1,237 @@
+    "-h",
+    type=click.STRING,
+    required=False,
+    default="localhost",
+    help="Hostname at which to serve. Default is localhost.",
+    envvar="DAGSTER_CODE_SERVER_HOST",
+)
+@click.option(
+    "--max-workers",
+    "-n",
+    type=click.INT,
+    required=False,
+    default=None,
+    help="Maximum number of (threaded) workers to use in the code server",
+)
+@python_origin_target_argument
+@click.option(
+    "--use-python-environment-entry-point",
+    is_flag=True,
+    required=False,
+    default=False,
+    help=(
+        "If this flag is set, the server will signal to clients that they should launch "
+        "dagster commands using `<this server's python executable> -m dagster`, instead of the "
+        "default `dagster` entry point. This is useful when there are multiple Python environments "
+        "running in the same machine, so a single `dagster` entry point is not enough to uniquely "
+        "determine the environment."
+    ),
+    envvar="DAGSTER_USE_PYTHON_ENVIRONMENT_ENTRY_POINT",
+)
+@click.option(
+    "--fixed-server-id",
+    type=click.STRING,
+    required=False,
+    help=(
+        "[INTERNAL] This option should generally not be used by users. Internal param used by "
+        "dagster to spawn a server with the specified server id."
+    ),
+)
+@click.option(
+    "--log-level",
+    type=click.Choice(["critical", "error", "warning", "info", "debug"], case_sensitive=False),
+    show_default=True,
+    required=False,
+    default="info",
+    help="Level at which to log output from the code server process",
+)
+@click.option(
+    "--container-image",
+    type=click.STRING,
+    required=False,
+    help="Container image to use to run code from this server.",
+    envvar="DAGSTER_CONTAINER_IMAGE",
+)
+@click.option(
+    "--container-context",
+    type=click.STRING,
+    required=False,
+    help=(
+        "Serialized JSON with configuration for any containers created to run the "
+        "code from this server."
+    ),
+    envvar="DAGSTER_CONTAINER_CONTEXT",
+)
+@click.option(
+    "--inject-env-vars-from-instance",
+    is_flag=True,
+    required=False,
+    default=False,
+    help="Whether to load env vars from the instance and inject them into the environment.",
+    envvar="DAGSTER_INJECT_ENV_VARS_FROM_INSTANCE",
+)
+@click.option(
+    "--location-name",
+    type=click.STRING,
+    required=False,
+    help="Name of the code location this server corresponds to.",
+    envvar="DAGSTER_LOCATION_NAME",
+)
+@click.option(
+    "--startup-timeout",
+    type=click.INT,
+    required=False,
+    default=180,
+    help="How long to wait for code to load or reload before timing out. Defaults to no timeout.",
+    envvar="DAGSTER_CODE_SERVER_STARTUP_TIMEOUT",
+)
+@click.option(
+    "--instance-ref",
+    type=click.STRING,
+    required=False,
+    help="[INTERNAL] Serialized InstanceRef to use for accessing the instance",
+    envvar="DAGSTER_INSTANCE_REF",
+)
+def start_command(
```

Comment:
some concern over ensuring all these settings get passed over to the proxy correctly